src/plugins/gs-plugin-eos.c

```diff
@@ -358,14 +359,23 @@ gs_plugin_eos_blacklist_kapp_if_needed (GsPlugin *plugin, GsApp *app)
 	last_token = tokens[num_tokens - 1];
 
 	if (!gs_plugin_locale_is_compatible (plugin, last_token)) {
+		g_debug ("Blacklisting '%s': incompatible with the current "
+			 "locale", gs_app_get_unique_id (app));
 		gs_app_add_category (app, "Blacklisted");
 		return TRUE;
 	}
 
 	locale_cache_key = get_app_locale_cache_key (app_name);
+	cached_app = gs_plugin_cache_lookup (plugin, locale_cache_key);
+
+	if (cached_app == app)
+		return FALSE;
```

Comment:
If you check this here, you can remove the same check from `gs_plugin_update_locale_cache_app`